### PR TITLE
Cars - CRUD 

### PR DESCRIPTION
**Pull request by yaseerokino**
### Reservation Backend API

**Hey Team**

_**In this milestone PR, I completed the following:**_

- Created the endpoint for PATCH, GET, DELETE and POST requests for cars r
- Update the User schema with a role column 
- Created method to allow only admin user create, update and delete a car

### CODE REVIEW

File: app/controllers/api/v1/cars_controller.rb [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
+
+  private
+
+  def set_car
+    @car = Car.find(params[:id])
+  end
+
+  def car_params
+    params.require(:car).permit(:name, :description, :price)
+  end
```
Comment (TimmyChan99) [inline app/controllers/api/v1/cars_controller.rb:50]: Good job there, please make sure to permit also the `reserved` parameter to allow the user to update the car status if it is reserved or not.

**Review by TimmyChan99 — CHANGES_REQUESTED** [review]
**Required Changes ♻️**
Hi @yaseerokino 

Good job so far!
There are some issues that you still need to work on to go to the next task but you are almost there!

**Comment by yaseerokino** [comment]
Hey @TimmyChan99 the reserved field should be updated automatically when the user reserves a car.

**Review by TimmyChan99 — APPROVED** [review]
**Status: Approved** 🟢
 
Hey @yaseerokino,

If the status already updated automatically then your project is complete you are ready to merge it   👍

**Review by Mwapsam — APPROVED** [review]

**Pull request merged**